reflex/components/radix/themes/color_mode.py

```diff
@@ -55,44 +57,87 @@ def create(
         )
 
 
-class ColorModeSwitch(Switch):
-    """Switch for toggling light / dark mode via toggle_color_mode."""
+LiteralPosition = Literal["top-left", "top-right", "bottom-left", "bottom-right"]
 
-    @classmethod
-    def create(cls, *children, **props):
-        """Create a switch component bound to color_mode.
+position_values = get_args(LiteralPosition)
 
-        Args:
-            *children: The children of the component.
-            **props: The props to pass to the component.
+position_map = {
+    "position": position_values,
+    "left": ["top-left", "bottom-left"],
+    "right": ["top-right", "bottom-right"],
+    "top": ["top-left", "top-right"],
+    "bottom": ["bottom-left", "bottom-right"],
+}
 
-        Returns:
-            The switch component.
-        """
-        return Switch.create(
-            *children,
-            checked=color_mode != LIGHT_COLOR_MODE,
-            on_change=toggle_color_mode,
-            **props,
-        )
 
+# needed to inverse contains for find
+def _find(const, var):
+    return Var.create_safe(const).contains(var)
 
-class ColorModeButton(Button):
-    """Button for toggling chakra light / dark mode via toggle_color_mode."""
+
+def _set_var_default(props, position, prop, default1, default2=""):
+    props.setdefault(
+        prop, cond(_find(position_map[prop], position), default1, default2)
+    )
+
+
+def _set_static_default(props, position, prop, default):
+    if prop in position:
+        props.setdefault(prop, default)
+
+
+class ColorModeIconButton(IconButton):
+    """Icon Button for toggling light / dark mode via toggle_color_mode."""
 
     @classmethod
-    def create(cls, *children, **props):
-        """Create a button component that calls toggle_color_mode on click.
+    def create(
+        cls,
+        *children,
+        position: LiteralPosition | None = None,
+        **props,
+    ):
+        """Create a icon button component that calls toggle_color_mode on click.
 
         Args:
             *children: The children of the component.
+            position: The position of the icon button. Follow document flow if None.
             **props: The props to pass to the component.
 
         Returns:
             The button component.
         """
-        return Button.create(
-            *children,
+        if children:
+            console.deprecate(
+                feature_name="passing children to color_mode.button",
+                reason=", use color_mode_cond and toggle_color_mode instead to build a custom color_mode component",
+                deprecation_version="0.5.0",
+                removal_version="0.6.0",
+            )
+
+        # position is used to set nice defaults for positioning the icon button
+        if isinstance(position, Var):
+            _set_var_default(props, position, "position", "fixed", position)
+            _set_var_default(props, position, "bottom", "2rem")
```

Comment:
might be nice to expose the `"2rem"` offset as a prop as well

Reply:
it's a set default so they can still override it by passing `top`, `bottom`, `left`, `right`